### PR TITLE
tests: improve naming

### PR DESCRIPTION
**Pull request by ntarocco**
On @fenekku feedback, this PR renames the test superuser and role to hopefully be more clear. Please add any comment.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -38,31 +38,31 @@ def create_app():
...
-    db.session.add(superusers)
-    db.session.add(ActionRoles.allow(superuser_access, role=superusers))
+def admin_role(db):
+    """Grant `superuser_access` action to the new admin role."""
+    admin = Role(name="admin")
+    db.session.add(admin)
+    db.session.add(ActionRoles.allow(superuser_access, role=admin))
     db.session.commit()
-    return superusers
+    return admin
```
Comment (fenekku) [inline tests/conftest.py:47]: Are admins and superusers the same thing? I thought someone who has access to the admin doesn't necessarily have the rights to do everything e.g., a library assistant (which has the right to access the admin panel) can come in to generate report from the admin panel and do metadata enhancements on records, but only superusers can also delete records and so on.

**Review by fenekku — COMMENTED** [review]
If admins and superusers are the same thing now, then this is fine. If they shouldn't be the same thing, then it's worth making the distinction.

Feel free to merge with explanations! (even if just a comment)

File: tests/conftest.py (hunk 1 above)
Comment (ntarocco) [reply tests/conftest.py:47]: It looks like indeed that there is a confusion of naming, in general. Invenio-administration has made the situation even worse.
What I want to highlight in the code is the difference between the `superuser-access` action, and any role names or users that will be allowed to perform the action. Developers are generally confused by that, and that's why I don't want to use the same name as the action.

To answer your question: it is up to the sys admin to create roles, without names that make sense in their domain. It could be `admin` or `superuser` or `librarion` or anything else.
The actions are, instead, hardcoded:
- [superusers-access](https://github.com/inveniosoftware/invenio-access/blob/master/invenio_access/permissions.py#L37)
- [administration-access](https://github.com/inveniosoftware/invenio-administration/blob/main/invenio_administration/permissions.py#L14)
You can allow any role you want to perform the actions above.

In short: it is very import to distinguish, in the code, the roles and the action. IMO, using the same naming, will create only confusion.

Comment (ntarocco) [reply tests/conftest.py:47]: No they are not: roles and actions are different, see my other comment.

**Review by fenekku — APPROVED** [review]
Thanks for the explanations. Let's see if I understood correctly:

- there is an `ActionNeed` (in-memory entity) `superuser_access` (with string `"superuser-access"`) which is the fundamental indicator of "superuserness" 
- there is a Role (db entity) with an arbitrary name
- there is an ActionRole (db entity) that ties that Role to the ActionNeed
- there is a RoleNeed (in-memory entity) which wraps the Role in a Need (in-memory flask-principal fundamental unit).

The same setup applies for `"administration-access"`.

The PermissionPolicy `needs` returns, via [_expand_action](https://github.com/inveniosoftware/invenio-access/blob/master/invenio_access/permissions.py#L180) deep in its call hierarchy, that `RoleNeed` for that Role tied to the `superuser_access` `ActionNeed` in its set of Needs.

If that's right, then then name `admin_role_need` lets me believe that it would be the `RoleNeed` for that `Role` tied to the `administration-access` need... :thinking: 

What about any of the following names: 

- `role_with_superuser_access` (respectively `role_with_superuser_access_identity`, `role_with_superuser_access_need`). 
- I also like to abbreviate `with` by `w` so `role_w_superuser_access` and so on
- `role_w_superuser` ...
- `role_of_superuser` ...

? Do you think one of them would help with the confusion in developers that you've observed where they conflate a role name that sounds like a superuser with that role giving superuser access on its own, rather than by the role  _being tied_ to the `superuser_access` `ActionNeed`? Maybe we can also just put your explanation for the name `admin` in the comment and basta. 

Naming is hard. Personally, the original `superuser_role` is fine by me because the potential conflation is ok: the Role that sounds like a superuser is actually in practice what determines superuser access (returned in `needs`) and the conftest sets it up that way. (there is a whole separate discussion about returning the raw `superuser_access` :laugh:). 

I leave it up to you - feel free to merge with the approach you think will work best :+1:

**Comment by ntarocco** [comment]
@fenekku I have applied your suggestions, thank you!
